[Fix] `jsx-no-target-blank`: allow rel to be an expression

odinho opened the pull request:
Previously this would be an error, since the rel wasn't read:

    <a target={"_blank"} href={"//"} rel={"noopener noreferrer"}>

Now this works, so you can enable this rule together with
`jsx-curly-brace-presence`.

Fixes #2508.

ljharb commented:
Since you made this PR from a fork that isn't your own, I'm unable to force push to it (due to a github bug that they're unlikely to fix because so few people do this); would you mind adding me to https://github.com/whereby/eslint-plugin-react so i can?

ljharb commented on tests/lib/rules/jsx-no-target-blank.js:
```diff
@@ -35,18 +35,23 @@ ruleTester.run('jsx-no-target-blank', rule, {
     {code: '<a href="foobar"></a>'},
     {code: '<a randomTag></a>'},
     {code: '<a target />'},
+    {code: '<a target={null}></a>'},
```
this isn't a valid `target` attribute value; it might even result in a runtime warning from react.

odinho commented:
Added you as "write" collaborator.

Updated tests, removed `target={null}` and added one `'` and one <code>`</code> test.

> Since you made this PR from a fork that isn't your own, I'm unable to force push to it (due to a github bug that they're unlikely to fix because so few people do this); would you mind adding me to https://github.com/whereby/eslint-plugin-react so i can?

Ouch. That feels weird. I'm an organization member on that one though. -- But I'll use my private user place next time :)
